Lib/http/client.py

```diff
@@ -309,9 +309,12 @@ def begin(self):
             if status != CONTINUE:
                 break
             # skip the header from the 100 response
+            header_total_size = 0
```

Comment:
rather than having this while loop doing header skipping on `100 Continue` responses here, it'd be better for us to just call `parse_headers(self.fp)` - ignoring the result.  That'll already take care of preventing an infinite loop as it obeys `_MAXHEADERS` in addition to `_MAXLINE` with a proper error for each condition and reduces the complexity of the code.

Reply:
Makes sense. Added a change. Also extract out the headers read part of `parse_headers` into `read_headers` to avoid unnecessary parse overhead in this case.